rstudio/shiny-server
# Use SockJS 1.5.0

This upgrades our SockJS client from a copy-and-pasted copy of SockJS 0.3.4, to a proper Node dependency version of v1.5.0. Among other benefits, I'll know when new versions of the sockjs-client are released, and any security vulnerabilities will be flagged.

The user-visible changes are limited to support for the new transports `eventsource` and `htmlfile`, which were introduced into the UI by shiny-server-client 1.2.0 but not actually supported by Shiny Server until now AFAIK. The new protocols will automatically be disabled if the older protocols `iframe-eventsource` and `iframe-htmlfile`, respectively, are disabled (by `disable_protocols`). This is to prevent breaking existing installations of Shiny Server that disable "everything but X".

Also, note that `htmlfile` doesn't work unless the Shiny app contains the line `<script>document.domain = document.domain;</script>`. I don't fully understand this but [it seems to be a bad idea](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy#changing_origin), maybe just pretend that protocol doesn't exist.

## Testing notes

* Ensure that `eventsource` and `htmlfile` appear in the protocol chooser UI and can be enabled.
* Ensure `eventsource` works (I know `htmlfile` doesn't work, see note about `document.domain` above).
* Ensure `disable_protocols streaming;` disables both `eventsource` and `htmlfile`.
* Ensure `disable_protocols iframe-eventsource;` disables `eventsource`.
* Ensure `disable_protocols iframe-htmlfile;` disables `htmlfile`.
